synapse/lib/ast.py

```diff
@@ -357,6 +357,48 @@ def __init__(self, kids=()):
         async for item in self.kids[0].run(runt, genr):
             yield item
 
+class InitBlock(AstNode):
+    '''
+    init {
+        // stuff here runs *once* before the first node yield (even if there are no nodes)
+    }
+    '''
+
+    async def run(self, runt, genr):
+
+        subq = self.kids[0]
+
+        once = False
```

Comment:
Seems like this could be just:
```
async for innr in subq.run(runt, agen()):
    pass
async for item in genr:
    yield item
```
instead of using once?